ledger/recovering-indexer-integration-tests/src/test/suite/scala/com/digitalasset/platform/indexer/RecoveringIndexerIntegrationSpec.scala

```diff
@@ -288,13 +289,23 @@ object RecoveringIndexerIntegrationSpec {
         materializer: Materializer,
         loggingContext: LoggingContext,
     ): ResourceOwner[ParticipantState] = {
-      val readWriteService = new PartyOnlyQueueWriteService(
-        ledgerId,
-        participantId,
-      )
-      ResourceOwner.successful(
-        readWriteService -> readWriteService
-      )
+      ResourceOwner
+        .forReleasable(() =>
```

Comment:
There is also `forBoundedSourceQueue`.

Reply:
Yes, but it's more difficult to make it work for this usecase. The given `forBoundedSourceQueue` has strong assumptions about the materialized value (expects a future) while here we have a source because of the integration with the broadcast hub